### PR TITLE
Adds Prettier

### PR DESCRIPTION
**Pull request by orta**
Code talks re: #618, I'm happy to let others make the call on whether this should happen - but this PR:

- Adds prettier to the repo
- Ensures that when you make a commit and JS files are updated
- Runs any related tests in Jest, bailing on any failure

This will fail CI, specifically because of one rule: [Require Following Curly Brace Conventions](http://eslint.org/docs/rules/curly) which would need to be turned to "multi" in exchange for prettier to handle it.

Turning off semi-colons is also possible, we can do it on the entire codebase then too. 

I ran `yyarn prettier -- --trailing-comma es5 --print-width 120   --no-semi --write "{lib,test,schema,integration}/**/**/**/**/*.js"` to generate the PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,7 +81,10 @@
     "expect.js": "^0.3.1",
     "flow-bin": "^0.37.0",
     "foreman": "^1.4.1",
+    "husky": "^0.13.3",
```
Comment (craigspaeth) [inline package.json]: ❤️

**Comment by alloy** [comment]
Epic 🎉 

On some level I do feel like it’s removing a certain personal touch 🤖 but I do like not having to think about it anymore 👍

**Comment by craigspaeth** [comment]
👍  I am huge fan of this with the caveat that I don't know what it's actually like working with it day to day 😛  I will say that I'd prefer to have mandatory argument parens, and no semicolons—but ultimately I'm happy to defer my personal style to the prettier overlords. Also I didn't know about husky and lint-staged!! There's a JS module for everything! This seems like an ideal way to do linting similar to Go's gofmt. Love it ❤️

**Comment by orta** [comment]
Yah - I'm a no semis, and use double quotes kind of human. (We can switch to double quote too, can do it all in one PR then) I also prefer the mandatory parens, but that isn't an option here alas, they really strive for no options.

```
$ "/Users/orta/dev/projects/artsy/js/sites/metaphysics/node_modules/.bin/prettier" --help
Usage: prettier [opts] [filename ...]

Available options:
  --write                  Edit the file in-place. (Beware!)
  --list-different or -l   Print filenames of files that are different from Prettier formatting.
  --stdin                  Read input from stdin.
  --print-width <int>      Specify the length of line that the printer will wrap on. Defaults to 80.
  --tab-width <int>        Specify the number of spaces per indentation-level. Defaults to 2.
  --use-tabs               Indent lines with tabs instead of spaces.
  --no-semi                Do not print semicolons, except at the beginning of lines which may need them.
  --single-quote           Use single quotes instead of double quotes.
  --no-bracket-spacing     Do not print spaces between brackets.
  --jsx-bracket-same-line  Put > on the last line instead of at a new line.
  --trailing-comma <none|es5|all>
                           Print trailing commas wherever possible. Defaults to none.
  --parser <flow|babylon>  Specify which parse to use. Defaults to babylon.
  --no-color               Do not colorize error messages.
  --version or -v          Print Prettier version.
```

**Comment by alloy** [comment]
What are mandatory parens?

**Comment by alloy** [comment]
Is it limiting lines to like 80 chars or something?

**Comment by alloy** [comment]
I’m also in favour of no semis and double-quotes btw.

**Comment by orta** [comment]
Him, I think that's what you're talking about yeah, I'm chill on that- I was more worried about seeing things like this:

```js
if(x)
  y();
else {
  y()
  z()
}
```

**Comment by vjeux** [comment]
> I was more worried about seeing things like this:

Prettier doesn't remove or add brackets. All it does is to format them. Your snippet looks like the following after prettier:

```js
if (x) y()
else {
  y()
  z()
}
```

https://prettier.github.io/prettier/#%7B%22content%22%3A%22if(x)%5Cn%20%20y()%3B%5Cnelse%20%7B%5Cn%20%20y()%5Cn%20%20z()%5Cn%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22%22%2C%22semi%22%3Afalse%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D

**Comment by orta** [comment]
[That is true](https://prettier.github.io/prettier/#%7B%22content%22%3A%22if(x)%7B%5Cn%20%20y()%3B%5Cn%7D%20else%20%7B%5Cn%20%20y()%5Cn%20%20z()%5Cn%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22%22%2C%22semi%22%3Afalse%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D) - thanks @vjeux 

I see what was happening, we had things like this:

```diff
-    if (!accessToken) return new Error('You need to be signed in to perform this action');
+     if (!accessToken)
+      return new Error('You need to be signed in to perform this action');
```

Where it allows single-lined un-bracket'd statement - but when it hit the line limit it went over to another line. I bet moving to 120 chars should fix them.

**Comment by alloy** [comment]
Alright, let’s land that puppy.

**Pull request merged**